src/puppetlabs/puppetdb/query_eng/engine.clj

```diff
@@ -1072,11 +1072,15 @@
                                             :value value})
 
                :number
-               (map->BinaryExpression {:operator :=
-                                       :column field
-                                       :value (if (string? value)
-                                                (ks/parse-number (str value))
-                                                value)})
+               (if (string? value)
```

Comment:
Below you have the `throw` come in the else clause, maybe an `if-not` here would be more consistent?